core/store/src/db.rs

```diff
@@ -381,6 +385,18 @@ impl Database for RocksDB {
                 DBOp::Delete { col, key } => unsafe {
                     batch.delete_cf(&*self.cfs[col as usize], key);
                 },
+                DBOp::DeleteAll { col } => {
+                    let cf_handle = unsafe { &*self.cfs[col as usize] };
+                    let opt_first = self.db.iterator_cf(cf_handle, IteratorMode::Start).next();
+                    let opt_last = self.db.iterator_cf(cf_handle, IteratorMode::End).next();
+                    assert_eq!(opt_first.is_some(), opt_last.is_some());
+                    if let (Some((min_key, _)), Some((max_key, _))) = (opt_first, opt_last) {
+                        if min_key != max_key {
+                            batch.delete_range_cf(cf_handle, &min_key, &max_key)
+                        }
```

Comment:
> Removes the database entries in the range ["begin_key", "end_key"), i.e., including "begin_key" and excluding "end_key". It is not an error if no keys exist in the range ["begin_key", "end_key").

https://docs.rs/rocksdb/0.15.0/rocksdb/struct.WriteBatch.html#method.delete_range_cf

1. It might be a good idea to put this comment into the code for future reference
2. `min_key != max_key` seem redundant, though it does not hurt